### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by szepeviktor**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Hello!
I am new here.
Spent an hour fixing all typos.
How should we proceed?

### CODE REVIEW

**Comment by szepeviktor** [comment]
`.typos.toml`
```toml
[files]
extend-exclude = [
    ".git/",
    "vendor/",
    "bpf/include/linux/",
    ".mailmap",
    "AUTHORS",
    "Documentation/spelling_wordlist.txt",
]
ignore-hidden = false

[default]
extend-ignore-re = [
    "Parth Patel",
    "JUMO",
    "[0-9A-Za-z+/=]{40,64}",
    "[0-9a-f]{10}\\.\\.[0-9a-f]{10}",
    "2c490643-dd95-4f73-8862-139afe771ffd",
    " -hda ",
    '"BA"',
    "stap's",
    "\\.rela\\.eh_frame",
    "graceful-term-svc-pg7nd",
    "br-22b28ede79c2",
]

[default.extend-words]
"aks" = "aks"
"etcp" = "etcp"
"mke" = "mke"
"unmarshaling" = "unmarshaling"
# bad Go naming
"ba" = "ba"
"fo" = "fo"
"nd" = "nd"
"fpr" = "fpr"

[default.extend-identifiers]
"filename_2nd_half" = "filename_2nd_half"
"ICMP6_ND_TARGET_OFFSET" = "ICMP6_ND_TARGET_OFFSET"
"ICMP6_ND_OPTS" = "ICMP6_ND_OPTS"
"ND_OPT_TARGET_LL_ADDR" = "ND_OPT_TARGET_LL_ADDR"
"PT_REGS_PARM1" = "PT_REGS_PARM1"
"ANDed" = "ANDed"
"output_2nd_file" = "output_2nd_file"
"ar_tha" = "ar_tha"
"del2nd" = "del2nd"
"del2ndWanted" = "del2ndWanted"
"ep2ndApply" = "ep2ndApply"
"k8sSvc2ndApply" = "k8sSvc2ndApply"
"upsert2nd" = "upsert2nd"
"upsert2ndWanted" = "upsert2ndWanted"
"iNdEx" = "iNdEx"
"L2_RESPONSER_MAP4_SIZE" = "L2_RESPONSER_MAP4_SIZE"
"VALUEs" = "VALUEs"
"poolBUID" = "poolBUID"
"serviceBUID" = "serviceBUID"
```

File: contrib/release/bump-readme.sh [hunk 1]
```diff
@@ -34,19 +34,19 @@ update_release() {
               | sed 's/^.*'"$obj_regex"'.*tag\/v\('"$rem_branch_regex"'\).*$/\1/')
     old_date=$(git log -1 -s --format="%cI" v$current | sed "$REGEX_FILTER_DATE")
     new_date=$(git log -1 -s --format="%cI" $latest | sed "$REGEX_FILTER_DATE")
-    elease=$(echo $old_branch | sed 's/v//')
+    release=$(echo $old_branch | sed 's/v//')
```
Comment (szepeviktor) [inline contrib/release/bump-readme.sh:37]: This in the function may overwrite the global `$release` variable.
In functions one can use `local`.

**Comment by learnitall** [comment]
Hey @szepeviktor! Thanks for the PR. This is super helpful stuff and I really appreciate the work you put into this. 

This PR is _massive_ as it changes files across the entire project and pulls in a reviewer for almost every code owner we have. PRs tend to be difficult and sluggish to merge with this many required reviewers since reviewers need to filter through all the changes for those that are relevant to them. Additionally, if something breaks in CI, it can be difficult to track down the source of the problem.

Can we work together to help structure your changes into smaller PRs? We can look at the CODEOWNERs file to figure out different logical sections of the repo to fix typos of. Also, I noticed the `.typos.toml` file that you posted. Are you using https://github.com/crate-ci/typos/tree/master for finding typos? I'd love to learn more about your workflow and it would be great if we could create a GitHub workflow to automate this process. We currently have a spell checker for documentation, but we don't have anything for the code base or commit messages.

I have the following recommendations for next steps, please let me know what you think:

1. Let's create an issue in the repository to track this effort.
2. Let's take a look at your branch and figure out how to split these changes into smaller PRS.
3. If you're up for it, let's automate your workflow into a GitHub workflow that we can add into the repository to check for typos in the future.

If you have questions, please feel free to reach out to me on the community slack (`@Ryan Drew`) or swing by the [community meeting](https://docs.cilium.io/en/latest/community/community/#weekly-community-meeting) next week.

**Comment by joestringer** [comment]
I'll convert this to draft for now while we figure out next steps. Initial CI smoke tests are also failing, so it's likely there are unintended breakages in the PR as-is.

**Comment by szepeviktor** [comment]
> If you're up for it, let's automate your workflow into a GitHub workflow that we can add into the repository to check for typos in the future.

That is easy.
typos has an action but it is not the nicest thing in the world when looking at it in detail: `uses: "crate-ci/typos@master"`
This is how I run typos in CI: https://github.com/szepeviktor/byte-level-care/blob/876446a09724872c78773de442459f53fc5d627f/.github/workflows/spelling.yml#L31-L50

**Pull request closed without merging**